portfolio/src/main/webapp/script.js

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Fetches content of /data from the server and adds it to the DOM.
+ */
+function getData() {
+
+  const responsePromise = fetch('/data');
+
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addDataToDom().
+ */
+function handleResponse(response) {
+
+  const textPromise = response.text();
+
+  textPromise.then(addDataToDom);
+}
+
+/** Adds data to the DOM. */
+function addDataToDom(data) {
+
+  const dataContainer = document.getElementById('data-container');
+  
+  dataContainer.innerText = data;
+}
+
+function getComments(){
+    fetch('/data').then(response => response.json()).then((comments) => {
+     const statsListElement = document.getElementById('data-container');
+     statsListElement.innerHTML = '';
+     statsListElement.appendChild(
```

Comment:
In this code, you're relying on knowing exactly how many comments the client will get back from the server, and this code could easily break if the server returns a different number of comments and this code isn't updated.

I think you're better off using a for loop here, to loop through all the comments you get from the server, and then append them one by one to statsListElement. That way you can handle any amount of comments you get from the server.

Reply:
+1 this is exactly the kind of thing we were discussing in this mornings pod sync. 

For example, imagine if it was the chatroom, even though we are starting with a single user, its better to use a collection (even if you KNOW for sure there is only 1 user), because the code will automatically work when updated with multiple.

In this case, you can use the es6->

comments.forEach((comment, index) {
// stats list append child thing
}